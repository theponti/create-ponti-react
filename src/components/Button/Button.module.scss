.wrap {
  background-color: rgb(0 133 255);
  border-radius: $shape_borderRadius;
  border: 0;
  color: rgb(255 255 255);
  cursor: pointer;
  font-size: 16px;
  min-height: 30px;
  padding: 12px 16px;

  &:disabled {
    opacity: 0.7;
    cursor: not-allowed;
  }

  &:hover,
  &:focus {
    background-color: $grey_light100;
    box-shadow: $focusShadow;
  }

  &.danger {
      color: $white;
      background-color: $red_main;
  }

  &.success {
    color: $white;
    background-color: $green_main;
  }

  &.text {
    color: $black;
    background-color: $white;
  }
}
