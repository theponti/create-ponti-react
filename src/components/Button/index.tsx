import cx from 'classnames';

import styles from './Button.module.scss';

enum VARIANTS {
  DANGER = 'danger',
  SUCCESS = 'success',
  TEXT = 'text',
}

function Button({
  children,
  className,
  onClick,
  variant,
}: ButtonType) {
  let buttonStyle;

  switch (variant) {
    case VARIANTS.SUCCESS:
      buttonStyle = styles.btnSuccess;
      break;
    case VARIANTS.DANGER:
      buttonStyle = styles.btnDanger;
      break;
    case VARIANTS.TEXT:
      buttonStyle = styles.btnText;
      break;
    default:
  }

  return (
    <button
      data-testid="button"
      className={cx(styles.container, className, buttonStyle)}
      onClick={onClick}
      type="button"
    >
      {children}
    </button>
  );
}

Button.defaultProps = {
  className: '',
  variant: VARIANTS.SUCCESS,
};

type ButtonType = {
  children: React.ReactNode;
  className?: string;
  onClick: () => void;
  variant?: 'danger' | 'success' | 'text';
};

export default Button;
