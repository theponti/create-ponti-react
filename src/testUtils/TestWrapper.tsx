import PropTypes, { InferProps } from "prop-types";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "services/store";

function TestWrapper({ children }: InferProps<typeof TestWrapper.propTypes>) {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
}

TestWrapper.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TestWrapper;
