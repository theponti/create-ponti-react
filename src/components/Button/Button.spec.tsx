import { render } from '@testing-library/react';
import TestWrapper from 'testUtils/TestWrapper';
import {
  describe, expect, it, vi,
} from 'vitest';
import Button from '.';

describe('<Button/>', () => {
  it('should render a text button', () => {
    const onClick = vi.fn();
    const { container } = render(
      <TestWrapper>
        <Button variant="text" onClick={onClick}>Button</Button>
      </TestWrapper>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render a success button', () => {
    const onClick = vi.fn();
    const { container } = render(
      <TestWrapper>
        <Button variant="success" onClick={onClick}>Button</Button>
      </TestWrapper>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render a danger button', () => {
    const onClick = vi.fn();
    const { container } = render(
      <TestWrapper>
        <Button variant="danger" onClick={onClick}>Button</Button>
      </TestWrapper>,
    );
    expect(container).toMatchSnapshot();
  });
});
