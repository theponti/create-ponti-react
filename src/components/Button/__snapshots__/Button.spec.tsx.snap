// Vitest Snapshot v1

exports[`<Button/> > should render a danger button 1`] = `
<div>
  <button
    class="_container_7jj5b_1 _btnDanger_7jj5b_16"
    data-testid="button"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> > should render a success button 1`] = `
<div>
  <button
    class="_container_7jj5b_1 _btnSuccess_7jj5b_21"
    data-testid="button"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> > should render a text button 1`] = `
<div>
  <button
    class="_container_7jj5b_1 _btnText_7jj5b_26"
    data-testid="button"
    type="button"
  >
    Button
  </button>
</div>
`;

exports[`<Button/> > should render component 1`] = `
<div>
  <button
    class="_container_7jj5b_1 _btnSuccess_7jj5b_21"
    data-testid="button"
    type="button"
  >
    Button
  </button>
</div>
`;
